validator/server_arb/execution_run.go

```diff
@@ -79,6 +83,96 @@ func (e *executionRun) GetStepAt(position uint64) containers.PromiseInterface[*v
 	})
 }
 
+func (e *executionRun) GetMachineHashesWithStepSize(machineStartIndex, stepSize, requiredNumHashes uint64) containers.PromiseInterface[[]common.Hash] {
+	return stopwaiter.LaunchPromiseThread(e, func(ctx context.Context) ([]common.Hash, error) {
+		return e.machineHashesWithStepSize(ctx, machineStartIndex, stepSize, requiredNumHashes)
+	})
+}
+
+func (e *executionRun) machineHashesWithStepSize(
+	ctx context.Context,
+	machineStartIndex,
+	stepSize,
+	requiredNumHashes uint64,
+) ([]common.Hash, error) {
+	if stepSize == 0 {
+		return nil, fmt.Errorf("step size cannot be 0")
+	}
+	if requiredNumHashes == 0 {
+		return nil, fmt.Errorf("required number of hashes cannot be 0")
+	}
+	machine, err := e.cache.GetMachineAt(ctx, machineStartIndex)
+	if err != nil {
+		return nil, err
+	}
+	log.Debug(fmt.Sprintf("Advanced machine to index %d, beginning hash computation", machineStartIndex))
+
+	// If the machine is starting at index 0, we always want to start at the "Machine finished" global state status
+	// to align with the machine hashes that the inbox machine will produce.
+	var machineHashes []common.Hash
+	if machineStartIndex == 0 {
```

Comment:
I don't understand what this is doing. It feels like it's purposefully pushing some hash that is different than all of the machine hashes that will follow. I read the comment above, but I still don't really understand what this is about.